src/Framework/Framework/Resources/Scripts/state-manager.ts

```diff
@@ -404,3 +404,21 @@ function createWrappedObservable<T>(initialValue: DeepReadonly<T>, typeHint: Typ
     });
     return obs
 }
+
+function compareObjectTypes(currentValue: any, newVal: any): boolean {
+    if (currentValue["$type"] && currentValue["$type"] === newVal["$type"]) {
+        // objects with type must have a same type
+        return true;
+    }
+    else if (!currentValue["$type"] && !newVal["$type"]) {
```

Comment:
don't dynamic objects have `$type: { type: "dynamic" }`?

Reply:
No. 
The `{ type: "dynamic" }` can appear only in type metadata, not in the ViewModel objects
In the ViewModel objects, the `$type` property is always `string` and is always a hash of the .NET type name, which is subsequently looked up in type metadata. Missing `$type` there means that it is a dynamic object.

Reply:
Ok.